nss-ivory-trolls/WorkforceManagement
# Computers

# Description

Added the following functionality:

**Given** a user wants to view all computers
**When** the user clicks on the Computers item in the navigation bar
**Then** the user should see a list of all computers
**And** each item should be a hyperlink that can be clicked to view the details

**Given**  a user is viewing all computers
**When** the user clicks the Create New link
**Then** the user should be presented with a form in which the following information can be entered

1. Computer manufacturer
2. Computer make
3. Purchase date

**Given**  user is viewing a single computer
**When** the user clicks on the Delete link
**Then** the user should be presented with a screen to verify that it should be deleted
**And** if the user chooses Yes from that screen, the computer should be deleted **only if it is has never been assigned to an employee**


## Type of change



# Testing Instructions for Change Made

```
git fetch --all
git checkout computers
dotnet run
```

1. Click on "Computers" navigation item and confirm a list of computers appears.
2. Click "Create New" link at the top of the list of computers and confirm form contains fields for Computer Manufacturer, Computer Maker and Purchase Date.
3. Enter test data into fields and submit. View should return to list of computers with the newly entered computer added.
4. Click the details link of a computer in the list and confirm details are displayed.
5. Click the delete link of the computer you just created. You should see a confirmation screen and a delete button. Delete the computer. View should return to list of computers and the computer that was deleted shoud not be in the list.
6. Click the delete link of another computer from the original data inserted into our database. The computer should be assigned an employee. The delete confirmation screen should notify you that the computer has been assigned to an employee and there should be no delete button.


# Checklist:
